text/3463-crates-io-policy-update.md

```diff
@@ -0,0 +1,233 @@
+
+You are responsible for using crates.io in compliance with all applicable laws,
+regulations, and all of our policies. These policies may be updated from time to
+time. We will interpret our policies and resolve disputes in favor of protecting
+users as a whole. The crates.io team reserves the possibility to evaluate each
+instance on a case-by-case basis.
+
+For issues such as DMCA violations, or trademark and copyright infringements,
+the crates.io team will respect the legal decisions of the
+[Rust Foundation](https://rustfoundation.org/) as the official legal entity
+providing the crates.io service.
+
+
+## Package Ownership
+
+crates.io has a first-come, first-serve policy on crate names. Upon publishing a
+package, the publisher will be made owner of the package on crates.io.
+
+If you want to take over a package, we require you to first try and contact the
+current owner directly. If the current owner agrees, they can add you as an
+owner of the crate, and you can then remove them, if necessary. If the current
+owner is not reachable or has not published any contact information the
+crates.io team may reach out to help mediate the process of the ownership
+transfer.
+
+The crates.io team only deletes crates from the registry that do not comply with
+the policies on this document. Crate deletion by their owners is not possible to
+keep the registry as immutable as possible. If you want to flag your crate as
+open for transferring ownership to others, you can publish a new version with
+a message in the README or description communicating to the crates.io support
+team that you consent to transfer the crate to the first person who asks for it:
+
+> I consent to the transfer of this crate to the first person who asks
+> help@crates.io for it.
+
+
+## Data Access
+
+If you need access to a large subset of the crates.io database we recommend
+first looking at the **crates.io [index repository](https://github.com/rust-lang/crates.io-index)**.
+This repository is updated live whenever new versions are published and contains
+all the information needed for cargo to run the dependency resolution algorithm.
+
+In case the index dataset is insufficient for your purposes, we also publish a
+**database dump** every 24 hours. This includes the majority of data from our
+database except for sensitive private information. The latest database dump is
+available at <https://static.crates.io/db-dump.tar.gz> and information on using
+the content is contained in the tarball. Please note that while we aim to keep
+the data structure somewhat stable, we can not give any stability guarantees on
+the exact database table layouts.
+
+If the index repository and the database dump are insufficient you may also use
+the crates.io API directly, though it is at the discretion of the crates.io to
+block any excessive usage. We require users of the crates.io API to limit
+themselves to a maximum of 1 request per second.
```

Comment:
This seems incorrect - does this need a carveout for the download-crate and sparse registry endpoint?

Reply:
Or perhaps a carve-out for usage via `cargo` itself?

Reply:
> does this need a carveout for the download-crate and sparse registry endpoint?

the sparse registry is not technically part of the API (crates.io/api/), but you are right that we should probably exclude the download endpoint. this section was mostly adapted from https://crates.io/policies#crawlers, but since the crawler context isn't as strong anymore such an exception makes sense.

thanks for the suggestion!

Reply:
https://github.com/rust-lang/crates.io/pull/7048 fixed this and a couple of other things. we'll merge the changes from that PR back into this RFC once/if the RFC gets accepted.